src/schematools/types.py

```diff
@@ -406,16 +426,25 @@ def __init__(
         if not self["schema"].get("$schema", "").startswith("http://json-schema.org/"):
             raise ValueError("Invalid JSON-schema contents of table")
 
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> DatasetTableSchema:
+        """Create tableschema from a dict.
+
+        When this factory function is used, the _parent_schema argument is missing
+        and will be set to an empty DatasetSchema.
+        """
+        return cls(DatasetSchema({"id": "parent", "type": "dataset"}), obj)
```

Comment:
Is dat handig? straks kunnen we debuggen waarom er zo'n vreemde "parent" dataset rondspookt.

Reply:
De id en parent waren optioneel, maar zijn eigenlijk overal nodig. Probleem als ze optioneel zijn is ook dat je een shitload extra checks nodig hebt om mypy tevreden te stellen.
Overigens wordt deze factory function voor de DatasetTable maar op 1 plek in de DSO gebruikt in een testje. We kunnen ook deze factory function weghalen. Die fake dataset zit mij idd ook niet lekker.

Reply:
This fake parent was not a good idea. I reverted it and made the parameters optional again.